src/containers/FileExplorer/FileUploadTab.tsx

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators, Dispatch } from 'redux';
+import { Asset, UploadFileMetadataResponse } from '@cognite/sdk';
+import { Select, Spin, Button, message } from 'antd';
+import styled from 'styled-components';
+import {
+  ExtendedAsset,
+  createNewAsset,
+  selectAssets,
+} from '../../modules/assets';
+import { RootState } from '../../reducers';
+import FileUploader from '../../components/FileUploader';
+import { trackSearchUsage } from '../../utils/metrics';
+import { sdk } from '../../index';
+
+const Wrapper = styled.div`
+  .wrapper {
+    margin-top: 16px;
+  }
+  button {
+    margin-top: 6px;
+  }
+`;
+type Props = {
+  assetId?: number;
+  asset?: ExtendedAsset;
+  onFileUploaded: (file: UploadFileMetadataResponse) => void;
+};
+
+type State = {
+  includeAssetId: number | undefined;
+  fetching: boolean;
+  searchResults: Asset[];
+};
+
+class FileUploadTab extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      includeAssetId: props.assetId,
+      fetching: false,
+      searchResults: props.asset ? [props.asset] : [],
+    };
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (this.props.asset !== prevProps.asset) {
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({
```

Comment:
Do you know how we can do this properly? There is probably a reason why you shouldn't set state in `componentDidUpdate`.

Reply:
^naw they just dont want recursive calls, its all in good interest, i can section out this code, but it makes it much less readable as only this function calls this

Reply:
this will never be a recursive call because prevProps.asset !== asset only if the selected asset (globally changed) and any of our setState does not affect that

Reply:
Yeah I know, so fine if it's thought through :)